core/trino-main/src/main/java/io/trino/sql/planner/plan/NonReadingTableExecuteNode.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.planner.plan;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+import io.trino.metadata.TableExecuteHandle;
+import io.trino.metadata.TableHandle;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.sql.planner.Symbol;
+
+import java.util.List;
+import java.util.Optional;
+
+public class NonReadingTableExecuteNode
```

Comment:
Let's call it more positively, like `SimpleTableExecuteNode` or `DataLessTableExecuteNode`, ...
cc @losipiuk @alexjo2144 for the naming.